src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -199,6 +213,33 @@ protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
             return base.UpdateValueCore(updateValueParams);
         }
 
+        /// <summary>
+        /// Updates the Script property of the node.
+        /// NOTE: This is a temporary method used during the Python 2 to Python 3 transistion period,
+        /// it will be removed when the transistion period is over.
+        /// </summary>
+        /// <param name="newCode">The new migrated code</param>
+        internal void MigrateCode(string newCode)
+        {        
+            var e = new PythonCodeMigrationEventArgs(Script, newCode); 
+            Script = newCode;
+            if(Engine  != PythonEngineVersion.CPython3)
+                Engine = PythonEngineVersion.CPython3;
```

Comment:
One thing I'm seeing, when the node is using IronPython2 and I migrate the code, is that the execution happens twice, first with a warning and then successfully. My suspicion is that at the point this line runs, an execution is triggered with the old code, causing this weird effect.

Reply:
This was happening as we where calling `UpdateScript` twice when migrating the code, form `OnEngineChanged` and `OnNodeModelCodeMigrated` i changed this so that `OnNodeModelCodeMigrated` doesnt call `UpdateScript` but instead changes the engine which calls `OnEngineChanged`. This seems to solve it.